api/api-template-config-v7/src/main/java/com/thoughtworks/go/apiv7/admin/templateconfig/TemplateConfigControllerV7.java

```diff
@@ -170,6 +173,16 @@ public String show(Request req, Response res) throws IOException {
         }
     }
 
+    public String showParameters(Request req, Response res) throws IOException {
+        PipelineTemplateConfig templateConfig = fetchEntityFromConfig(req.params("template_name"));
+        if (isGetOrHeadRequestFresh(req, templateConfig)) {
```

Comment:
Here, we're checking for the ETag generated based on the entire template config, whereas, this API is only for the template parameters.

Imagine there is a change in the template config, but the template parameters hasn't changed. Where, the API would respond `412` even if there is no change in the response of the API.

---

Here are the options we've:
1. Compute ETag only for template parameters.
2. No ETag for this API?? we always send `200`?
3. Keep the implementation as is (i.e. rely on entire template config for etag).

@kritika-singh3 @ketan @arvindsv @maheshp @bdpiparva -- Thoughts??

Reply:
IMO, Etag should be generated based on the payload or entity being serialised. Generating based on entire config does not make sense.

Reply:
I was of the opinion that the parameters don't exist out of template. They can change with any change in the template config.
Scenarios I considered:
1. Parameters being added/removed.
2. The place where the parameters are utilized being changed.

So, decided to do a Etag check on the whole config. If anything, I would like to go with Option 2 as we are fetching the entire template config either way.

Reply:
@kritika-singh3 - Agreed, that parameters does not exist outside of  template config or can't be updated directly though API. I am leaning towards option 1 as the value of parameters can remain same when other parts of the template config is changed which defeats the purpose of having etag.

Reply:
@GaneshSPatil @bdpiparva I have updated the API to calculate eTag only for the parameters and using the same to send either `304` or `200`